tests/api/runtimes/test_kubejob.py

```diff
@@ -141,6 +148,304 @@ def test_run_with_node_selection(self, db: Session, client: TestClient):
             expected_affinity=affinity,
         )
 
+    def test_change_preemptible_modes_between_runs(
+        self, db: Session, client: TestClient
+    ):
+        node_selector = self._generate_node_selector()
+        mlrun.mlconf.preemptible_nodes.node_selector = base64.b64encode(
+            json.dumps(node_selector).encode("utf-8")
+        )
+        runtime = self._generate_runtime()
+        runtime.with_preemption_mode(PreemptionModes.prevent)
+        self._execute_run(runtime)
+        expected_anti_affinity = k8s_client.V1Affinity(
+            node_affinity=k8s_client.V1NodeAffinity(
+                required_during_scheduling_ignored_during_execution=k8s_client.V1NodeSelector(
+                    node_selector_terms=compile_anti_affinity_by_label_selector_no_schedule_on_matching_nodes(),
+                ),
+            ),
+        )
+        self.assert_node_selection(affinity=expected_anti_affinity)
+        # set anti affinity (simulates runtime that had prevent mode)
+        runtime.with_node_selection(affinity=expected_anti_affinity)
```

Comment:
I'm not sure how easy it is with the current infrastructure but ideally you should have a test with one runtime object on which you assign a preemption mode and the rest of the field are being assigned by the actual logic that does that, without you needing to mock that, even if you mock that, I prefer you to do that using the actual logic from the backend